pyhf/paramsets.py

```diff
@@ -53,11 +52,13 @@ def reduce_paramset_requirements(paramset_requirements):
 
     for param_name in list(paramset_requirements.keys()):
         params = paramset_requirements[param_name]
+        param_user_configs = paramsets_user_configs.get(param_name, {})
 
         combined_param = {}
         for param in params:
             for k in param_keys:
-                combined_param.setdefault(k, set([])).add(param.get(k))
```

Comment:
seems like if we reorder the loops here, we can only have one loop over `param_keys`?

Reply:
We can't. The first loop is iterating over all requirements and merging them all into a single (combined) requirement. After this loop is done, we look at the combined paramset to see if things are merge-able.